src/component/RestaurantListItem.js

```diff
@@ -0,0 +1,46 @@
+import React from "react";
+import { Link } from "react-router-dom";
+import PropTypes from "prop-types";
+import { constants } from "../constants";
+
+ /**
+  * Display restaurant information in a list format.
+  * @returns {JSX.Element}
+  */
+function RestaurantListItem (props) {
+    const { address, name, selected, slug, thumbnail } = props;
+
+    function detailsClickHandler() {
+        selected(props);
+    }
```

Comment:
I find there is a general lack of arrow functions for short hand overall, but especially here it feels like going out of our way **not** to write an arrow function `onClick={()=>selected(props)}`

Also, maybe we need to follow the convention of prefacing callbacks with `on*` when passing into components:
`onClick={()=> onSelected(props)}`

...also do we need to pass all the `props` here?